src/ESM.sol

```diff
@@ -66,67 +52,26 @@ contract ESM is DSNote {
         require(z <= x);
     }
 
-    // -- admin --
-    function file(bytes32 job, address obj) external auth note {
-        if (job == "end") end = EndLike(obj);
-        if (job == "sun") sun = obj;
-    }
-
-    function file(bytes32 job, uint256 val) external auth note {
-        if (job == "cap") cap = val;
-    }
-
-    // -- state changes --
     function fire() external note {
-        require(state == START && full(), "esm/not-fireable");
+        require(!fired, "esm/already-fired");
+        require(sum >= min, "esm/min-not-reached");
```

Comment:
Why did you move away from calling `full` here?

Reply:
Because I wanted to make `full() external`, for the reasons Lev brought up some time ago, namely that it makes the bytecode easier to audit/FV than `public`.

I know it looks weird, so I'm open to removing `fuil`.

Reply:
There's not much difference in the difficulty of FVing `public` vs. `external` functions

Reply:
OK I have no clue about this, but @livnev seemed to be into the idea.

I'd love to stop cargo-culting! :D

Reply:
@asymmetric @MrChico My thinking is along the lines of: (i) change `public` to `external` whenever possible, at the very least to reduce bytecode complexity; (ii) in some cases, if you really need to call it internally, consider changing `public` to `internal`, since then it will be easier to FV the calling methods using the internal method's spec as a lemma; (iii) in some cases, consider removing the method altogether.

Applying the analysis to this situation makes me want to ask: do we need `full()` at all?

Reply:
@livnev `external` vs. `public` only makes a difference for functions whose arguments cannot be stored directly on the stack, i.e. for dynamic types. In those cases, an `external` function can keep the arguments in `calldata` while a `public` function will have to copy them to `memory`. For value types, the difference between an `public` and `external` function are fairly equivalent.

Reply:
@MrChico I agree. There is also a general theme of trying to limit the number of entry points to your contract and making it so that a piece of code is not executed as part of too many paths (since this increases the overhead of reasoning about correctness). In this simple case I agree that it doesn't make a difference one way or the other.

Reply:
OK, seems like switching back to `public` is OK in this case.